scalatest/src/main/scala/org/scalatest/EitherValues.scala

```diff
@@ -98,6 +107,32 @@ trait EitherValues {
    */
   implicit def convertRightProjectionToValuable[L, R](rightProj: Either.RightProjection[L, R])(implicit pos: source.Position): RightValuable[L, R] = new RightValuable(rightProj, pos)
 
+  class EitherValuable[L, R](either: Either[L, R], pos: source.Position) {
+    /**
+      * Returns the <code>Left</code> value contained in the wrapped <code>LeftProjection</code>, if defined as a <code>Left</code>, else throws <code>TestFailedException</code> with
+      * a detail message indicating the <code>Either</code> was defined as a <code>Right</code>, not a <code>Left</code>.
+      */
+    def leftValue: L = {
+      try {
+        either.left.get
```

Comment:
well, you are basically doing the same thing, which is deprecated and might disappear in newer Scala versions (2.14? 3.0?).
I though the intent was to do
```
either.swap.getOrElse(throw new TestFailedException(...))
```
without `try` block.

Reply:
Only `.right.get` is deprecated because `Either` is now right biased by default.  The `.left` projections have been deprecated shortly but un-deprecated in https://github.com/scala/scala/pull/8012 :)

Reply:
But the `swap` followed by `getOrElse` is a bit nicer, so I used that now instead, thanks for the suggestion!